config-model/src/test/java/com/yahoo/config/model/provision/ModelProvisioningTest.java

```diff
@@ -1034,6 +1074,50 @@ public void testRequiringMoreNodesThanAreAvailable() {
         tester.createModel(services, false);
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void testRequiredNodesAndDedicatedClusterControllers() {
+        String services =
+                "<?xml version='1.0' encoding='utf-8' ?>\n" +
+                "<services>" +
+                "  <content version='1.0' id='foo'>" +
+                "     <redundancy>1</redundancy>" +
+                "     <documents>" +
+                "       <document type='type1' mode='index'/>" +
+                "     </documents>" +
+                "     <nodes count='2' required='true'/>" +
+                "  </content>" +
+                "</services>";
+
+        int numberOfHosts = 4; // needs 2 for foo and 3 for cluster controllers.
```

Comment:
4 is not equal to 5, so what happens here?

Reply:
That's why the exception is expected.

Reply:
D'oh 🤦